src/main/java/com/openshift/jenkins/plugins/pipeline/OpenShiftDeployer.java

```diff
@@ -46,105 +49,169 @@ public String getDepCfg() {
 	public String getWaitTime() {
 		return waitTime;
 	}
-	
-	public String getWaitTime(Map<String, String> overrides) {
+
+	public String getWaitUnit() {
+		if(waitUnit == null || waitUnit.length() == 0) {
+			if(waitTime.length() == 0) {
+				return "sec";
+			}
+
+			// For previous versions of the plugin, the default unit was milliseconds.
+			return "milli";
+		}
+
+		return waitUnit;
+	}
+
+	public String getWaitTime(Map<String,String> overrides) {
 		String val = getOverride(getWaitTime(), overrides);
 		if (val.length() > 0)
 			return val;
-		return Long.toString(getDescriptor().getWait());
+		return getDescriptor().getWait() + getDescriptor().getWaitUnit();
+	}
+
+	// Overridden for better type safety.
+	// If your plugin doesn't really define any property on Descriptor,
+	// you don't have to do this.
+	@Override
+	public DescriptorImpl getDescriptor() {
+		return (DescriptorImpl)super.getDescriptor();
+	}
+
+	/**
+	 * Descriptor for {@link OpenShiftDeployer}. Used as a singleton.
+	 * The class is marked as public so that it can be accessed from views.
+	 *
+	 */
+	@Extension // This indicates to Jenkins that this is an implementation of an extension point.
+	public static final class DescriptorImpl extends BuildStepDescriptor<Builder> {
+		private String wait;
+		private String waitUnit;
+		/**
+		 * To persist global configuration information,
+		 * simply store it in a field and call save().
+		 *
+		 * <p>
+		 * If you don't want fields to be persisted, use <tt>transient</tt>.
+		 */
+
+		/**
+		 * In order to load the persisted global configuration, you have to
+		 * call load() in the constructor.
+		 */
+		public DescriptorImpl() {
+			load();
+
+			if(waitUnit == null || waitUnit.length() == 0) {
```

Comment:
With perhaps a generic method in GlobalConfig that takes a parameter like the outer class name so that it knows to call deploy wait units vs. build wait units